### ISSUE TITLE
Corrupted chunk in save file leading to crash

### ISSUE THREAD

**Issue by malnaur** [anegostudios/VintageStory-Issues#3471]
### Game Version

1.19.0

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

Running on 64 bit Windows 10.0.22621.0 with 32452 MB RAM
Game Version: v1.19.0 (Stable)
1/18/2024 4:44:45 PM: Critical error occurred
Loaded Mods: butcheringfix@1.0.0, plantfix@, satup@0.1.0, game@1.19.0, autoloot@1.0.1, blocksoverlay@2.5.0, glowingprojectiles@1.1.3, hudclock@3.4.0, vmetp@1.1.0, vpeatp@1.1.0, vresp@1.1.0, creative@1.19.0, survival@1.19.0
System.OverflowException: Arithmetic operation resulted in an overflow.
   at Vintagestory.Common.ArrayConvert.ByteToByte(Byte[] data, Int32 length) in VintagestoryLib\Common\Convert\ArrayConvert.cs:line 73
   at Vintagestory.Common.CompressionZSTD.Decompress(Byte[] fi, Int32 offset, Int32 length) in VintagestoryLib\Common\Convert\Compression.cs:line 684
   at Vintagestory.Common.Compression.DecompressCombined(Byte[] blocksCompressed, Int32[][]& blocks, Int32& refCount, Func`1 newArray) in VintagestoryLib\Common\Convert\Compression.cs:line 208
   at Vintagestory.Common.ChunkDataLayer.Decompress(Byte[] layerCompressed) in VintagestoryLib\Common\Model\ChunkDataLayer.cs:line 755
   at Vintagestory.Common.ChunkData.DecompressFrom(Byte[] blocksCompressed, Byte[] lightCompressed, Byte[] lightPaletteCompressed, Byte[] fluidsCompressed, Int32 chunkdataVersion) in VintagestoryLib\Common\Model\ChunkData.cs:line 348
   at Vintagestory.Common.WorldChunk.unpackNoLock() in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 287
   at Vintagestory.Common.WorldChunk.UnpackAndReadBlock(Int32 index, Int32 layer) in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 223
   at Vintagestory.GameContent.ChunkMapLayer.GenerateChunkImage(Vec2i chunkPos, IMapChunk mc, Int32[]& tintedImage, Boolean colorAccurate) in VSEssentials\Systems\WorldMap\ChunkLayer\ChunkMapLayer.cs:line 493
   at Vintagestory.GameContent.ChunkMapLayer.OnOffThreadTick(Single dt) in VSEssentials\Systems\WorldMap\ChunkLayer\ChunkMapLayer.cs:line 344
   at Vintagestory.GameContent.WorldMapManager.<OnLvlFinalize>b__22_0()

### Logs

```Log```

**Comment by Craluminum2413** [comment]
@malnaur Does it happen without any mods installed?

**Comment by malnaur** [comment]
Turned off all client mods. Plantfix is my quick mod to remove references to lighter foliage. Still crashes shortly after start:

Running on 64 bit Windows 10.0.22621.0 with 32452 MB RAM
Game Version: v1.19.0 (Stable)
1/18/2024 4:49:09 PM: Critical error occurred
Loaded Mods: plantfix@, satup@0.1.0, game@1.19.0, creative@1.19.0, survival@1.19.0
System.OverflowException: Arithmetic operation resulted in an overflow.
   at Vintagestory.Common.ArrayConvert.ByteToByte(Byte[] data, Int32 length) in VintagestoryLib\Common\Convert\ArrayConvert.cs:line 73
   at Vintagestory.Common.CompressionZSTD.Decompress(Byte[] fi, Int32 offset, Int32 length) in VintagestoryLib\Common\Convert\Compression.cs:line 684
   at Vintagestory.Common.Compression.DecompressCombined(Byte[] blocksCompressed, Int32[][]& blocks, Int32& refCount, Func`1 newArray) in VintagestoryLib\Common\Convert\Compression.cs:line 207
   at Vintagestory.Common.ChunkDataLayer.Decompress(Byte[] layerCompressed) in VintagestoryLib\Common\Model\ChunkDataLayer.cs:line 756
   at Vintagestory.Common.ChunkData.DecompressFrom(Byte[] blocksCompressed, Byte[] lightCompressed, Byte[] lightPaletteCompressed, Byte[] fluidsCompressed, Int32 chunkdataVersion) in VintagestoryLib\Common\Model\ChunkData.cs:line 332
   at Vintagestory.Common.WorldChunk.unpackNoLock() in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 282
   at Vintagestory.Common.WorldChunk.UnpackAndReadBlock(Int32 index, Int32 layer) in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 220
   at Vintagestory.GameContent.ChunkMapLayer.GenerateChunkImage(Vec2i chunkPos, IMapChunk mc, Int32[]& tintedImage, Boolean colorAccurate) in VSEssentials\Systems\WorldMap\ChunkLayer\ChunkMapLayer.cs:line 493
   at Vintagestory.GameContent.ChunkMapLayer.OnOffThreadTick(Single dt) in VSEssentials\Systems\WorldMap\ChunkLayer\ChunkMapLayer.cs:line 322
   at Vintagestory.GameContent.WorldMapManager.<OnLvlFinalize>b__22_0() in VSEssentials\Systems\WorldMap\WorldMapManager.cs:line 206

**Comment by Craluminum2413** [comment]
@malnaur You still have mods, remove **all** of them

**Comment by malnaur** [comment]
Nope. Without plantfix, the game will crash due to the lighterfoliage bug. Satup just adjusts some sat settings for dairy. both on the server side which is NOT crashing. I did remove satup since that is really client side and the crash still happens. One more detail: this seems to be associated with teleporting in though I teleported in/out this location multiple times without a problem.

**Comment by radfast** [comment]
This is a deep game engine issue. Plantfix mod is OK here, that can't be the cause.

Most likely cause is a corrupted chunk internally somehow. It's fairly hard to understand how there can be a corrupted chunk on the client but not the server, may be a networking issue.

@malnaur strange question, but what region/language is your PC set to  (I don't mean the Vintage Story game settings language, I mean Windows).  And same question for the server?

**Comment by malnaur** [comment]
> This is a deep game engine issue. Plantfix mod is OK here, that can't be the cause.
> 
> Most likely cause is a corrupted chunk internally somehow. It's fairly hard to understand how there can be a corrupted chunk on the client but not the server, may be a networking issue.
> 
> @malnaur strange question, but what region/language is your PC set to (I don't mean the Vintage Story game settings language, I mean Windows). And same question for the server?

Both are US English.
While the issue was happening, restarting the client would briefly spawn in and then crash again. After a small number of these iterations (2 or 3?), the server would refuse connections and need a restart. If I managed to quickly tp away, the client would not crash, only to do so on return. As I said above, after turning on corrupt chunk regen on the server and and walking into the area from a nearby wp, I have not had a repeat of the crashes. I didn't see any obvious sign of regen as the items I placed at the destination (a few of the new traps) were as i left them. Of course, the regen could have run on a nearby chunk or otherwise invisibly. The new release does not seem all that stable, frankly.

**Comment by radfast** [comment]
TL;DR: there was a corrupted chunk in the save game and you have now fixed this using corrupt chunk regen?   Sounds like everything is working as intended.

**Comment by malnaur** [comment]
Except the corrupted chunk should not have happened in the first place.

**Comment by radfast** [comment]
The only possible causes of this type of corrupted chunk which I can think of are (a) hard shutdown of the game or the whole PC, exactly at the time when it is writing the save file to disk; (b) occasional rare errors in your RAM or disk system or other hardware or firmware (including maybe things like anti-virus software if it filters disk writes).   Things like that can happen with computers.  It's always possible that someone turns the power off.

Some time we ought to make it so that if a corrupted chunk like this is detected, it automatically regenerates it instead of crashing.  But this issue has been extremely rare in the game, so far, and it is fixable by running a command as you did, that is the currently intended resolution.
